isce2_topsapp/iono_proc.py

```diff
@@ -0,0 +1,383 @@
+                                    {get_burst(str(ifg))}')
+                # Get the swath and burst number
+                swath = get_swath(str(ifg))
+                burst = get_burst(str(ifg))
+                # Get mask
+                mask_file = workdir / f'mask/{swath}/msk_{burst}.rdr.vrt'
+                mask_ds = gdal.Open(str(mask_file), gdal.GA_ReadOnly)
+                # Mask
+                mask_interferogram(str(ifg), mask_ds.ReadAsArray())
+                mask_ds = None  #close
+                pbar.update()
+
+    # Step-3 Compute ionospheric correction
+    iono_xml = template.render(**iono_kwargs,
+                               iono_startStep=IONO_STEPS[1],
+                               iono_stopStep=IONO_STEPS[-1])
+
+    with open('ionoApp.xml', "w") as file:
+        file.write(iono_xml)
+
+    tops_app_cmd = f'{isce_application_path}/topsApp.py'
+
+    step_cmd = f'{tops_app_cmd} ionoApp.xml --dostep=ion'
+    result = subprocess.run(step_cmd,
+                            shell=True)
+    if result.returncode != 0:
+        raise ValueError('TopsApp failed at step: ion-rawion2esd')
+
+    # Step-4 mergeBursts
+    # Create merged/topophase.ion file
+    merge_bursts(range_looks=range_looks, azimuth_looks=azimuth_looks)
+
+    # Step-5 Geocode ionospheric correction outputs
+    step_cmd = f'{tops_app_cmd} ionoApp.xml --dostep=geocode'
+    result = subprocess.run(step_cmd,
+                            shell=True)
+    if result.returncode != 0:
+        raise ValueError('TopsApp failed at step: geocode (ion)')
+
+
+def merge_bursts(range_looks: int = 19,
+                 azimuth_looks: int = 7,
+                 ion_rangeLooks: int = 200,
+                 ion_azimuthLooks: int = 50,
+                 considerBursts: bool = False,
+                 mergedir: Union[str, Path] = './merged') -> None:
+    from isce.components.isceobj.TopsProc.runMergeBursts import interpolateDifferentNumberOfLooks
+
+    mergedIfgname = 'topophase.flat'
+    mergedIonname = 'topophase.ion'
+    #########################################
+
+    if considerBursts:
+        print('TODO, Skip as it is not a DEFAULT')
+        '''
+        ionDirname = 'ion/ion_burst'
+        topsProc : runMergeBursts.py 776
+        mergeBursts2(frames, os.path.join(ionDirname, 'IW%d',  'burst_%02d.ion'),\
+              burstIndex, box, os.path.join(mergedir, mergedIonname+suffix), virtual=virtual, validOnly=True)
+        multilook(os.path.join(mergedir, mergedIonname+suffix),
+                      outname = os.path.join(mergedir, mergedIonname),
+                      alks = self.numberAzimuthLooks, rlks=self.numberRangeLooks)
+        '''
+    else:
+        ionFilt = 'ion/ion_cal/filt.ion'
+        img = isceobj.createImage()
+        img.load(ionFilt+'.xml')
+        ionFiltImage = (np.fromfile(ionFilt, dtype=np.float32).reshape(img.length*2, img.width))[1:img.length*2:2, :]
```

Comment:
Could you please add a comment as to why this indexing is happening and/or link to the code in ISCE2 you are using?

Reply:
added link to the original isce2 script